iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -0,0 +1,356 @@
+        )
+
+        arches_str = ', '.join(sorted(arches))
+        log.debug('Set to regenerate the bundle image for the following arches: %s', arches_str)
+
+        payload = {
+            'from_bundle_image_resolved': from_bundle_image_resolved,
+            'state': 'in_progress',
+            'state_reason': f'Regenerating the bundle image for the following arches: {arches_str}',
+        }
+        exc_msg = 'Failed setting the resolved "from_bundle_image" on the request'
+        update_request(request_id, payload, exc_msg=exc_msg)
+
+        # Pull the from_bundle_image to ensure steps later on don't fail due to registry timeouts
+        podman_pull(from_bundle_image_resolved)
+
+        with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
+            manifests_path = os.path.join(temp_dir, 'manifests')
+            _copy_files_from_image(from_bundle_image_resolved, '/manifests', manifests_path)
+            metadata_path = os.path.join(temp_dir, 'metadata')
+            _copy_files_from_image(from_bundle_image_resolved, '/metadata', metadata_path)
+            new_labels = _adjust_operator_bundle(
+                manifests_path, metadata_path, organization, pinned_by_iib
+            )
+
+            with open(os.path.join(temp_dir, 'Dockerfile'), 'w') as dockerfile:
+                dockerfile.write(
+                    textwrap.dedent(
+                        f"""\
+                            FROM {from_bundle_image_resolved}
+                            COPY ./manifests /manifests
+                            COPY ./metadata /metadata
+                        """
+                    )
+                )
+                for name, value in new_labels.items():
+                    dockerfile.write(f'LABEL {name}={value}\n')
+
+            for arch in sorted(arches):
+                _build_image(temp_dir, 'Dockerfile', request_id, arch)
+                _push_image(request_id, arch)
+
+    set_request_state(request_id, 'in_progress', 'Creating the manifest list')
+    output_pull_spec = _create_and_push_manifest_list(request_id, arches)
+
+    conf = get_worker_config()
+    if conf['iib_index_image_output_registry']:
+        old_output_pull_spec = output_pull_spec
+        output_pull_spec = output_pull_spec.replace(
+            conf['iib_registry'], conf['iib_index_image_output_registry'], 1
+        )
+        log.info(
+            'Changed the bundle_image pull specification from %s to %s',
+            old_output_pull_spec,
+            output_pull_spec,
+        )
+
+    payload = {
+        'arches': list(arches),
+        'bundle_image': output_pull_spec,
+        'state': 'complete',
+        'state_reason': 'The request completed successfully',
+    }
+    update_request(request_id, payload, exc_msg='Failed setting the bundle image on the request')
+
+
+def _apply_package_name_suffix(metadata_path, organization=None):
```

Comment:
Do we want to get rid of `_` at the beginning of the name of the function?

Reply:
I don't think so because it's still a private function being used only in the same module.